### ISSUE TITLE
Local constants seen as global constants resulting in a duplicate symbols error

### ISSUE THREAD

**Issue by nic0mic** [mParticle/mparticle-apple-sdk#100]
Hi, 
### Issue found
We are using mParticle + different Kits + other 3rd SDKs.
Updating to the version 7.12.6 we got a `duplicate symbol '_version'` linker error.
It's a conflict between a mParticle kit file (in this case it was [this Adobe file](https://github.com/mparticle-integrations/mparticle-apple-integration-adobe/blob/master/mParticle-Adobe/MPIAdobe.m)) and another 3rd party SDK file.

### Why?
mParticle and Kits define Objective-C constants to be used in their code.
See the [Adobe kit code](https://github.com/mparticle-integrations/mparticle-apple-integration-adobe/blob/master/mParticle-Adobe/MPIAdobe.m):

Local constants are defined locally to the implementation `.m` like this:
```objective-c
NSString *version = @"2";
```
At compile time it is working (compiler checks only header `.h`), but at linking time the variables in implementation `.m`are also checked.
As we are using other SDKs, they have defined the same constant, with the same name, and they are both globally visible by the linker. Here the error `duplicate symbol`.

### How to fix?
Declare them as local constants using `static` and `const` keyword:
Change
```objective-c
NSString *version = @"2";
```
to
```objective-c
static NSString *const version = @"2";
```

In this way, they are local to the file and compilation unit. It fixed our issue but until a mParticle fix, we will have to edit the file at each update.

**Notes:**
- We got the issue with the `mparticle-apple-integration-adobe` repository but it could happen with the main code or other kits. It will be good to have this fix for each repository.
- Some constants were unused in the `MPIAdobe.m`file
- We could also add a prefix to the local constant but the real issue won't be fixed. (it's fine for the global constants) 

**Reference**
https://stackoverflow.com/questions/17228334/what-is-the-best-way-to-create-constants-in-objective-c

Thank you :)

**Comment by samdozor** [comment]
Hi @nic0mic - thank you for the comprehensive and well written issue! We will get this resolved in our next release - either this week or next.

**Comment by nic0mic** [comment]
@samdozor Thank you :D I won't forget to check the next update 👍

**Comment by samdozor** [comment]
hey @nic0mic this is now released as version 7.12.7. Closing for now but please reopen if the issue is persisting.

**Comment by nic0mic** [comment]
Hello,

The fix didn't work for us. There are 2 `MPIAdobe.m` files and we are still using the previous(?) one.😅
**[Not updated:](https://github.com/mparticle-integrations/mparticle-apple-integration-adobe/blob/7.12.7/mParticle-Adobe/MPIAdobe.m)** `mParticle-Adobe/MPIAdobe.m`
**[Updated:](https://github.com/mparticle-integrations/mparticle-apple-integration-adobe/blob/7.12.7/mParticle-Adobe-Media/MPIAdobe.m)** `mParticle-Adobe-Media/MPIAdobe.m`, in **Media** directory

We are still using Adobe kit without the Media dependency (ie via `pod 'mParticle-Adobe/Adobe')`
We got other issues with the `Media` dependency because we already had the `VideoHeartbeatLibrary` SDK added manually in our project. But this is another story ;)

I hope it is good for you to update the other file or let me know. 
Thanks
